### ISSUE TITLE
E2532 Parameters in step function tasks

### ISSUE THREAD

**Issue by davidpricedev** [awslabs/cfn-python-lint#520]
*cfn-lint version: (`cfn-lint --version`)*
0.10.4

*Description of issue.*
I used the step-function sample project called "[Transfer Data Records](https://docs.aws.amazon.com/step-functions/latest/dg/sample-project-transfer-data-sqs.html)" to  create a  step function json document. I plugged that into an existing CF template in place of the previous json. After filtering out the expected errors (referenced lambdas not existing and such) I'm left with 3 unexpected errors (line 616 is the 1st line of the json step-function document): 

>E2532 State Machine Definition key (Parameters) for State (Read Next Message from DynamoDB) of Type (Task) is not valid
>test.yaml:616:7
>
>E2532 State Machine Definition key (Parameters) for State (Send Message to SQS) of Type (Task) is not valid
>test.yaml:616:7
>
>E2532 State Machine Definition key (Parameters) for State (Pop Element from List) of Type (Pass) is not valid
> test.yaml:616:7

The json must be correct since it works and was written by AWS. This would seem to imply that the linter is complaining where it shouldn't.

**Comment by kddejong** [comment]
Looks like Parameters was added as an optional item for Tasks.  However I can't find any documentation confirming Parameters being added to Pass.
```
"Pop Element from List": {
   "Type": "Pass",
   "Parameters": {
     "List.$": "$.List[1:]"
   },
   "Next": "For Loop Condition"
},
```

Were you able to confirm this template works.

https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-pass-state.html

**Comment by davidpricedev** [comment]
I tried a few things along those lines, but wasn't able to find a way to accomplish the same thing that is happening with this task from that sample: 

```
                "Pop Element From List": {
                  "Type": "Pass",
                  "Parameters": {
                    "List.$": "$.List[1:]"
                  },
                  "Next": "For Loop Condition"
                },
```

**Comment by kddejong** [comment]
Since we added the Parameters part to the `Task` type are you ok with closing this?  If we can't make the `Parameters` part of the `Pass` work we should probably submit an issue against the documentation.  Thoughts?

**Comment by ericallam** [comment]
I have a state machine that works with the `Parameters` key in a state of type `Pass`:

```json
"Initialize Manifest": {
  "Type": "Pass",
  "Next": "Get Available Episodes",
  "Parameters": {
    "id.$": "$.workflowId"
  },
  "ResultPath": "$.manifest"
}
```
